File: src/main/java/project/rpg/commands/FriendCommand.java
package project.rpg.commands;

import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import project.rpg.player.info.Friend;

import java.util.ArrayList;
import java.util.List;

public class FriendCommand implements CommandExecutor, TabCompleter {
    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (sender instanceof Player) {
            Player player = (Player) sender;

            if(args.length==2) {
                switch (args[0]) {
                    case "add":
                        Friend.addFriend(player,args[1]);
                        break;
                    case "accept":
                        Friend.acceptFriend(player,args[1]);
                        break;
                    case "list":
                        Friend.printFriend(player);
                        break;
                    default:
                        //sender.sendMessage(command.getUsage());
                }
            } else if(args.length == 1 && args[0].equals("list")) {
                sender.sendMessage(Friend.getFriends().toString());
            }
        }
        return false;
    }

    @Override
    public @Nullable List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        List<String> result = new ArrayList<>();

        if(args.length == 1){
            result.add("add");
            result.add("accept");
            result.add("list");
        }

        if(args.length == 2 && args[0].equals("add")){
            for(Player player : Bukkit.getOnlinePlayers()){
                result.add(player.getName());
            }
        }

        return result;
    }
}

File: src/main/java/project/rpg/commands/FriendCommand.kt
package project.rpg.commands

import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter
import org.bukkit.entity.Player
import project.rpg.extensions.friends

class FriendCommand : CommandExecutor, TabCompleter {

    private val recommendations = listOf(listOf("add", "accept", "list"))

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (sender is Player) {
            val friends = sender.friends

            if (args.isNotEmpty() && args[0] == "list") {
                friends.printFriendsList()
                return true
            }
            if (args.size >= 2) {
                return when (args[0]) {
                    "add" -> friends.requestFriend(args[1])
                    "accept" -> friends.acceptFriend(args[1])
                    else -> false
                }
            }
        }
        return false
    }

    override fun onTabComplete(
        sender: CommandSender,
        command: Command,
        label: String,
        args: Array<String>
    ): List<String> {
        if (args.size == 1) {
            return recommendations[0]
        }
        if (sender is Player && args.size == 2) {
            val result: MutableList<String> = ArrayList()
            when (args[0]) {
                "add" -> Bukkit.getOnlinePlayers().forEach { result.add(it.name) }
                "accept" -> sender.friends.pendingList.forEach { result.add(it.name) }
            }
            return result
        }
        return emptyList()
    }
}